libs/libcommon/src/libcommon/operations.py

```diff
@@ -2,41 +2,89 @@
-from libcommon.utils import Priority
+from huggingface_hub.hf_api import DatasetInfo, HfApi
 
+from libcommon.exceptions import DatasetInBlockListError
+from libcommon.orchestrator import remove_dataset, set_revision
+from libcommon.storage_client import StorageClient
+from libcommon.utils import Priority, SupportStatus, raise_if_blocked
 
-def backfill_dataset(
-    dataset: str,
-    revision: str,
-    cache_max_days: int,
-    blocked_datasets: list[str],
-    priority: Priority = Priority.LOW,
-) -> None:
-    """
-    Update a dataset
 
-    Args:
-        dataset (str): the dataset
-        revision (str): The revision of the dataset.
-        cache_max_days (int): the number of days to keep the cache
-        priority (Priority, optional): The priority of the job. Defaults to Priority.LOW.
+@dataclass
+class DatasetStatus:
+    dataset: str
+    revision: str
+    support_status: SupportStatus
 
-    Returns: None.
 
-    Raises the following errors:
-        - [`libcommon.exceptions.DatasetInBlockListError`]
-          If the dataset is in the list of blocked datasets.
+def is_blocked(
+    dataset: str,
+    blocked_datasets: Optional[list[str]] = None,
+) -> bool:
+    if not blocked_datasets:
+        logging.debug("No datasets block list.")
+        return False
+    try:
+        raise_if_blocked(dataset, blocked_datasets)
+        logging.debug(f"Dataset is not blocked. Block list is: {blocked_datasets}, dataset is: {dataset}")
+        return False
+    except DatasetInBlockListError:
+        logging.debug(f"Dataset is blocked. Block list is: {blocked_datasets}, dataset is: {dataset}")
+        return True
+
+
+def get_support_status(
+    dataset_info: DatasetInfo,
+    blocked_datasets: Optional[list[str]] = None,
+) -> SupportStatus:
+    """
+      blocked_datasets (list[str]): The list of blocked datasets. Supports Unix shell-style wildcards in the dataset
+    name, e.g. "open-llm-leaderboard/*" to block all the datasets in the `open-llm-leaderboard` namespace. They
+    are not allowed in the namespace name.
     """
-    logging.debug(f"backfill {dataset=} {revision=} {priority=}")
-    DatasetOrchestrator(dataset=dataset, blocked_datasets=blocked_datasets).set_revision(
-        revision=revision, priority=priority, error_codes_to_retry=[], cache_max_days=cache_max_days
+    return (
+        SupportStatus.UNSUPPORTED
+        if (
+            is_blocked(dataset_info.id, blocked_datasets)
+            # ^ TODO: should we check here?
+            # As we don't keep the track of unsupported datasets,
+            # we lose the ability to send a meaningful error message to the user, as previously done:
+            #   "This dataset has been disabled for now. Please open an issue in"
+            #   " https://github.com/huggingface/datasets-server if you want this dataset to be supported."
+            # A solution is to double-check if the dataset is blocked in the API routes.
```

Comment:
sounds like a simple solution

Reply:
I'll do it in another PR.

Reply:
Done here: https://github.com/huggingface/datasets-server/pull/2246